### PR TITLE
`avm/res/dev-ops-infrastructure/pool`: removed preview API and added daysData UDT for user-friendly input

### PR DESCRIPTION
**Pull request by johnlokerse**
## Description

This pull requests:

- Updates the API version to non-preview
- Adds a user-defined type for daysData to make the input user-friendly
- Updates WAF deployment test
- Adds `formattedDaysData` to convert user input to actual input required by API

Tagging module owners: @elizatargithub7 @surajguptha

## Pipeline Reference

| Pipeline |
| -------- |
|      [avm.res.dev-ops-infrastructure.pool](https://github.com/johnlokerse/bicep-registry-modules/actions/workflows/avm.res.dev-ops-infrastructure.pool.yml)    |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [x] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/dev-ops-infrastructure/pool/README.md
Comment (AlexanderSehr) [inline avm/res/dev-ops-infrastructure/pool/README.md:1]: Please note, the `main.json` templates **must** be re-build with the latest Bicep version `0.31.34` as this version (running on the agents) renders a different `main.json` file then previous versions. 

The steps to execute would be
- Upgrade your Bicep CLI version to `0.31.34` (e.g. via `choco upgrade bicep` if choco is used)
- Re-run the `Set-AVMModule` script on your module. E.g. via `Set-AVMModule -ModuleFolderPath '.\avm\res\automation\automation-account\' -SkipFileAndFolderSetup -SkipReadMe -Recurse`
- Push the changes
- Re-run the tests to ensure everything works. 

If the static tests continue to fail for you, please note that there is currently presumably another new bug which affects some of our modules ([ref](https://github.com/Azure/bicep/issues/15517)).

Comment (johnlokerse) [reply avm/res/dev-ops-infrastructure/pool/README.md:1]: Hi @AlexanderSehr, I thought my Bicep was updated 😬... Will commit the updated main.json later 👍

Comment (johnlokerse) [reply avm/res/dev-ops-infrastructure/pool/README.md:1]: I just checked my commit and I see that the version is already at 0.31.34 or am I missing something? 😄

image

**Review by elizatargithub7 — DISMISSED** [review]

File: avm/res/dev-ops-infrastructure/pool/README.md
Comment (AlexanderSehr) [reply avm/res/dev-ops-infrastructure/pool/README.md:1]: @johnlokerse, all good, sorry. I literally commented all 30+ PRs to make sure it's not missed. You were just too fast 😄 Will resolve

File: avm/res/dev-ops-infrastructure/pool/main.bicep [hunk 1]
```diff
@@ -501,3 +524,40 @@ type managedIdentitiesType = {
   @description('Optional. The resource ID(s) to assign to the resource.')
   userAssignedResourceIds: string[]?
 }?
+
+type standbyAgentsConfigType = {
```
Comment (AlexanderSehr) [inline avm/res/dev-ops-infrastructure/pool/main.bicep:528]:
```suggestion
@export()
type standbyAgentsConfigType = {
```

File: avm/res/dev-ops-infrastructure/pool/main.bicep [hunk 2]
```diff
@@ -501,3 +524,40 @@ type managedIdentitiesType = {
...
+  endTime: string
+
+  @description('Required. The number of agents needed at the start time.')
+  startAgentCount: int
+
+  @description('Required. The number of agents needed at the end time.')
+  endAgentCount: int
+}?
+
+type daysDataType = {
```
Comment (AlexanderSehr) [inline avm/res/dev-ops-infrastructure/pool/main.bicep:542]:
```suggestion
@export()
type daysDataType = {
```

Requires a regeneration of the `main.json` file

**Comment by johnlokerse** [comment]
Added the exports @AlexanderSehr 👍

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**